tests/components/application-connector/internal/testkit/test-api/oauth.go

```diff
@@ -0,0 +1,138 @@
+package test_api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+type OAuthCredentials struct {
+	ClientID     string
+	ClientSecret string
+}
+
+const (
+	clientIDKey     = "client_id"
+	clientSecretKey = "client_secret"
+	grantTypeKey    = "grant_type"
+)
+
+type OauthResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+	Scope       string `json:"scope"`
```

Comment:
This field [is not required](https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2) and we do not handle it correctly, so maybe we should omit it